qa/tasks/cephfs/test_volumes.py

```diff
@@ -7652,6 +7652,107 @@ def test_subvolume_snapshot_clone_with_no_wait_not_enabled(self):
         # verify trash dir is clean
         self._wait_for_trash_empty()
 
+    def test_bulk_deletion_of_cancelled_clones(self):
+        subvolume = self._gen_subvol_name()
+        snapshot = self._gen_subvol_snap_name()
+        clones = self._gen_subvol_clone_name(6)
+
+        # create subvolume
+        self._fs_cmd("subvolume", "create", self.volname, subvolume, "--mode=777")
+
+        # do some IO
+        self._do_subvolume_io(subvolume, number_of_files=200)
+
+        # snapshot subvolume
+        self._fs_cmd("subvolume", "snapshot", "create", self.volname, subvolume, snapshot)
+
+        # Insert delay at the beginning of snapshot clone
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_delay', 15)
+
+        for i in range(0,3):
+            # schedule a clone
+            self._fs_cmd("subvolume", "snapshot", "clone", self.volname, subvolume, snapshot, clones[i])
+
+            # cancel pending clone
+            self._fs_cmd("clone", "cancel", self.volname, clones[i])
+
+            # check clone status
+            clone1_result = self._get_clone_status(clones[i])
+            self.assertEqual(clone1_result["status"]["state"], "canceled")
+            self.assertEqual(clone1_result["status"]["failure"]["errno"], "4")
+            self.assertEqual(clone1_result["status"]["failure"]["error_msg"], "user interrupted clone operation")
+
+        for i in range(3,6):
+            # schedule a clone
+            self._fs_cmd("subvolume", "snapshot", "clone", self.volname, subvolume, snapshot, clones[i])
+
+            # wait for clone to be in-progress
+            self._wait_for_clone_to_be_in_progress(clones[i])
+
+            # cancel in-progess clone
+            self._fs_cmd("clone", "cancel", self.volname, clones[i])
+
+            # check clone status
+            clone1_result = self._get_clone_status(clones[i])
+            self.assertEqual(clone1_result["status"]["state"], "canceled")
+            self.assertEqual(clone1_result["status"]["failure"]["errno"], "4")
+            self.assertEqual(clone1_result["status"]["failure"]["error_msg"], "user interrupted clone operation")
+
+        # clone removal should succeed with force after cancelled, remove clone1
+        self._fs_cmd("cancelled", "clones", "rm", self.volname, subvolume, "--force")
+
+        # remove snapshot
+        self._fs_cmd("subvolume", "snapshot", "rm", self.volname, subvolume, snapshot)
+
+        # remove subvolumes
+        self._fs_cmd("subvolume", "rm", self.volname, subvolume)
+
+        # verify trash dir is clean
+        self._wait_for_trash_empty()
+
+    def test_bulk_cancellation_of_pending_clones(self):
```

Comment:
I don't see any completed clones in these tests. You should make sure to have a completed clone before running this, and that it doesn't get damaged or throw any errors.